src/EventStore.Core/Services/SubscriptionMetric.cs

```diff
@@ -0,0 +1,74 @@
+#nullable enable
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using EventStore.Core.Messages;
+
+namespace EventStore.Core.Services;
+
+public class SubscriptionMetric {
+	private readonly ConcurrentDictionary<Guid, MonitoringMessage.SubscriptionStats> _subscriptionStats;
+
+	public SubscriptionMetric(Meter meter, string name) {
+		_subscriptionStats = new();
+
+		meter.CreateObservableUpDownCounter($"{name}-completed", Completed);
+		meter.CreateObservableUpDownCounter($"{name}-subscription-position", SubscriptionPosition);
+		meter.CreateObservableUpDownCounter($"{name}-stream-position", StreamPosition);
```

Comment:
the ticket talks about having the timestamps of the current vs latest events, is that still in scope?

Reply:
If we use the read index we could read the last event to get that info, but dunno how expensive that is every `n` seconds. We would have to expose `IAllReader` to do it for the all stream

Or we go back to using `EventCommitted`

Or, we could occasionally update this from the grpc subscription